File: CS/TextEditPopUp/TextEditPopUp.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.XtraEditors;
using System.ComponentModel;
using System.Windows.Forms;
using DevExpress.XtraEditors.Controls;

namespace TEPopUp
{
    class TextEditPopUp : PopupContainerEdit
    {
        static TextEditPopUp() { RepositoryItemTextEditPopUp.RegisterTextEditPopUp(); }
        public TextEditPopUp() : base() { }
        public override string EditorTypeName { get { return RepositoryItemTextEditPopUp.TextEditPopUpName; } }

        [DesignerSerializationVisibility(DesignerSerializationVisibility.Content)]
        public new RepositoryItemTextEditPopUp Properties
        { get { return base.Properties as RepositoryItemTextEditPopUp; } }

        public virtual object DataSource
        {
            set { Properties.DataSource = value; }
            get { return Properties.DataSource; }
        }

        public virtual string DisplayMember
        {
            set { Properties.DisplayMember = value; }
            get { return Properties.DisplayMember; }
        }
    }
}

File: VB/TextEditPopUp/TextEditPopUp.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.XtraEditors
Imports System.ComponentModel
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.Controls

Namespace TEPopUp
	Friend Class TextEditPopUp
		Inherits PopupContainerEdit
		Shared Sub New()
			RepositoryItemTextEditPopUp.RegisterTextEditPopUp()
		End Sub
		Public Sub New()
			MyBase.New()
		End Sub
		Public Overrides ReadOnly Property EditorTypeName() As String
			Get
				Return RepositoryItemTextEditPopUp.TextEditPopUpName
			End Get
		End Property

		<DesignerSerializationVisibility(DesignerSerializationVisibility.Content)> _
		Public Shadows ReadOnly Property Properties() As RepositoryItemTextEditPopUp
			Get
				Return TryCast(MyBase.Properties, RepositoryItemTextEditPopUp)
			End Get
		End Property

		Public Overridable Property DataSource() As Object
			Set(ByVal value As Object)
				Properties.DataSource = value
			End Set
			Get
				Return Properties.DataSource
			End Get
		End Property

		Public Overridable Property DisplayMember() As String
			Set(ByVal value As String)
				Properties.DisplayMember = value
			End Set
			Get
				Return Properties.DisplayMember
			End Get
		End Property
	End Class
End Namespace
